File: time.py
import time
import os
import sys

h = 0
m = 0
s = 0


def clear_screen():
    os.system("cls" if os.name == "nt" else "clear")


def start(lim):
    global h, m, s

    while True:
        clear_screen()

        print(f"CLOCK : {h:02d}:{m:02d}:{s:02d}")

        time.sleep(1)

        s += 1

        if s == 60:
            s = 0
            m += 1

        if m == 60:
            m = 0
            h += 1

        if h == 24:
            h = 0

        # Ask the user every 'lim' seconds
        if s % lim == 0:
            print("\n1. Continue")
            print("2. Stop")
            print("3. Restart")
            print("4. Exit")

            ch = int(input("\nEnter your choice: "))

            if ch == 1:
                continue

            elif ch == 2:
                print(f"\nClock Stopped at {h:02d}:{m:02d}:{s:02d}")
                time.sleep(2)
                return

            elif ch == 3:
                h = 0
                m = 0
                s = 0

                print("\nClock Restarted!")
                time.sleep(1)

            elif ch == 4:
                print("Program Terminated.")
                sys.exit()

            else:
                print("Invalid Choice!")
                time.sleep(1)


def main():
    while True:
        print("\n===== DIGITAL CLOCK =====")

        lim = int(input("Enter limit: "))

        print("1. Start Clock")
        print("2. Exit")

        opt = int(input("Enter your option: "))

        if opt == 1:
            start(lim)

        elif opt == 2:
            print("Program Terminated.")
            break

        else:
            print("Invalid Option!")


main()

File: time.c
#include <stdio.h>
#include <stdlib.h>
#include <windows.h>

struct clock
{
    int h;
    int m;
    int s;
};

struct clock t = {0, 0, 0};

void start(int lim)
{
    int ch;

    while (1)
    {
        system("cls");

        printf("CLOCK : %02d:%02d:%02d\n", t.h, t.m, t.s);

        Sleep(1000);

        t.s++;

        if (t.s == 60)
        {
            t.s = 0;
            t.m++;
        }

        if (t.m == 60)
        {
            t.m = 0;
            t.h++;
        }

        if (t.h == 24)
        {
            t.h = 0;
        }

        if (t.s % lim == 0)
        {
            printf("\n\n1. Continue");
            printf("\n2. Stop");
            printf("\n3. Restart");
            printf("\n4. Exit");

            printf("\n\nEnter your choice: ");
            scanf("%d", &ch);

            switch (ch)
            {
                case 1:
                    /* Continue running */
                    continue;

                case 2:
                    printf("\nClock Stopped at %02d:%02d:%02d\n",
                           t.h, t.m, t.s);
                    Sleep(2000);
                    return;

                case 3:
                    t.h = 0;
                    t.m = 0;
                    t.s = 0;

                    printf("\nClock Restarted!");
                    Sleep(1000);
                    break;

                case 4:
                    exit(0);

                default:
                    printf("\nInvalid Choice!");
                    Sleep(1000);
            }
        }
    }
}

int main()
{
    int opt;
    int lim;

    while (1)
    {
        
        printf("\n===== DIGITAL CLOCK =====\n");
        printf("enter limit\n");
        scanf("%d",&lim);
        printf("\n1. Start Clock");
        printf("\n2. Exit");
        printf("\nEnter your option: ");

        scanf("%d", &opt);

        switch (opt)
        {
            case 1:
                start(lim);
                break;

            case 2:
                printf("Program Terminated.\n");
                return 0;

            default:
                printf("Invalid Option!\n");
        }
    }

    return 0;
}
